zcash_primitives/src/transaction/mod.rs

```diff
@@ -715,6 +863,12 @@ impl Transaction {
             .map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "valueBalance out of range"))
     }
 
+    fn read_note_value<R: Read>(mut reader: R) -> io::Result<NoteValue> {
```

Comment:
`orchard.rs` looks like a better place for this

Reply:
definitely